svc/FetchService.js

```diff
@@ -50,44 +72,54 @@ export class FetchService {
      * @returns {Promise<Response>} - Promise which resolves to a Fetch Response.
      */
     async fetch(opts) {
-        let {params, method, contentType, url, autoAbortKey} = opts;
+        let {params, method, headers, url, autoAbortKey} = opts;
         throwIf(!url, 'No url specified in call to fetchService.');
+        warnIf(opts.contentType, 'contentType has been deprecated - please pass a "Content-Type" header instead.');
 
         // 1) Compute / install defaults
         if (!method) {
             method = (params ? 'POST' : 'GET');
         }
 
-        if (!contentType) {
-            contentType = (method === 'POST') ? 'application/x-www-form-urlencoded': 'text/plain';
-        }
-
         if (!url.startsWith('/') && !url.includes('//')) {
             url = XH.baseUrl + url;
         }
 
-        // 2) Prepare merged options
+        // 2) Compute headers
+        if (!(headers instanceof Headers)) {
+            const baseHeaders = {
+                'Content-Type': (method === 'POST') ? 'application/x-www-form-urlencoded': 'text/plain'
+            };
+            if (opts.acceptJson) {
+                baseHeaders['Accept'] = 'application/json';
+                delete opts.acceptJson;
+            }
+
+            const {defaultHeaders} = this;
+            headers = new Headers(Object.assign(
+                baseHeaders,
+                isFunction(defaultHeaders) ? defaultHeaders() : defaultHeaders,
+                isPlainObject(headers) ? headers : {}
+            ));
+        }
+        delete opts.headers;
```

Comment:
Ah - I see before we weren't mutating opts (we cleaned up the merged `fetchOpts` instead), but now we are in two places with headers and acceptJson.  I can't think of any instances of an app re-using / holding a reference to the opts passed in here, but still as a general practice best to leave them alone and might be a bit of a code smell that the logic around building up the merged / defaulted opts could be clarified.

Reply:
Can we just delete line 83?